storage/shard.go

```diff
@@ -128,6 +129,9 @@ type dbShard struct {
 	newSeriesBootstrapped    bool
 	ticking                  bool
 	shard                    uint32
+
+	// Only used in tests
+	testListIterBatchSize int
```

Comment:
this feels really ugly, should we just drop that parameter from the test?

Reply:
We could, however it is kind of the point though of the test I think (to fix off by one/length of array issues).  I don't feel too strongly one way or another, I can drop it.

Reply:
could we hide it away using fn injection, something like `shard.iterateBatchSizeFn`?